datalake/src/main/java/com/sequenceiq/datalake/flow/verticalscale/diskupdate/handler/DatalakeDiskUpdateValidationHandler.java

```diff
@@ -0,0 +1,101 @@
+            ResourceV4Response resource = stackV4Response.getResources().stream()
+                    .filter(res -> null != res.getInstanceId() && res.getInstanceGroup().equals(diskUpdateRequest.getGroup())
+                            && res.getResourceType().equals(ResourceType.AWS_VOLUMESET)).toList().get(0);
+            VolumeSetAttributes volumeSetAttributes = new Json(resource.getAttributes()).get(VolumeSetAttributes.class);
+            LOGGER.debug("Checking if the attached volumes have size less than the requested volumes.");
+            List<VolumeSetAttributes.Volume> attachedVolumes = volumeSetAttributes.getVolumes().stream()
+                    .filter(volume -> volume.getSize() < diskUpdateRequest.getSize()).collect(Collectors.toList());
+            boolean requestedSizeGreaterThanAvailable = attachedVolumes.size() > 0;
+            boolean diskTypeChangeSupported = datalakeV4Endpoint.isDiskTypeChangeSupported(stackV4Response.getCloudPlatform().toString());
+            if (!requestedSizeGreaterThanAvailable || !diskTypeChangeSupported) {
+                String exceptionMessage = !diskTypeChangeSupported ? "Disk Type Change/ Resize not supported for the requested platform" :
+                        String.format("Requested size for %s group is lesser than available", diskUpdateRequest.getGroup());
+                Exception ex = new Exception("Validation Failed: " + exceptionMessage);
+                DatalakeDiskUpdateFailedEvent failedEvent =
+                        new DatalakeDiskUpdateFailedEvent(payload, ex, DatalakeStatusEnum.DATALAKE_DISK_UPDATE_VALIDATION_FAILED);
+                LOGGER.error("Disk update failed with validation error: {} on stack {}", ex.getMessage(), payload.getStackCrn());
+                eventSender().sendEvent(failedEvent, datalakeDiskUpdateEvent.getHeaders());
+            } else {
+                DatalakeDiskUpdateEvent diskUpdateEvent = DatalakeDiskUpdateEvent.builder()
+                        .withAccepted(new Promise<>())
+                        .withResourceCrn(payload.getResourceCrn())
+                        .withResourceId(payload.getResourceId())
+                        .withResourceName(payload.getResourceName())
+                        .withDiskUpdateRequest(payload.getDatalakeDiskUpdateRequest())
+                        .withStackCrn(payload.getStackCrn())
+                        .withClusterName(payload.getClusterName())
+                        .withAccountId(payload.getAccountId())
+                        .withSelector(DatalakeDiskUpdateStateSelectors.DATALAKE_DISK_UPDATE_EVENT.selector())
+                        .withVolumesToBeUpdated(attachedVolumes)
+                        .withCloudPlatform(stackV4Response.getCloudPlatform().toString())
+                        .build();
+                eventSender().sendEvent(diskUpdateEvent, datalakeDiskUpdateEvent.getHeaders());
+            }
```

Comment:
move this logic to CB side

Reply:
No - all this is related to the datalake. so it must stay in datalake. but the services that need to be reused for datahubs are actually implemented in core side. only the handler is in the datalake side